File: backend/src/main/java/com/efedotov/vaultly/dto/auth/RegisterRequest.java
package com.efedotov.vaultly.dto.auth;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import jakarta.validation.constraints.Size;
import lombok.Data;

@Data
public class RegisterRequest {
    @NotBlank
    @Size(min = 3, max = 50)
    @Pattern(regexp = "^[a-zA-Z0-9_.-]+$", message = "Username: a-z, A-Z, 0-9, _, ., -")
    private String username;

    @NotBlank
    @Size(min = 12, max = 64, message = "Пароль должен быть от 12 до 64 символов")
    @Pattern(regexp = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\\d)(?=.*[^A-Za-z0-9]).+$", message = "Пароль должен содержать строчную, заглавную буквы, цифру и спецсимвол")
    private String password;

    private String firstName;
    private String lastName;
    private String publicKey;
    private String privateKeyEncrypted;
    private String salt;

    @Email
    @Size(max = 255)
    private String email;
}

File: lib/server/model/auth/register_request.dart
class RegisterRequest {
  final String username;
  final String password;
  final String? firstName;
  final String? lastName;
  final String? publicKey;
  final String? privateKeyEncrypted;
  final String? salt;
  final String? email;

  RegisterRequest({
    required this.username,
    required this.password,
    this.firstName,
    this.lastName,
    this.publicKey,
    this.privateKeyEncrypted,
    this.salt,
    this.email,
  });

  factory RegisterRequest.fromJson(Map<String, dynamic> json) {
    return RegisterRequest(
      username: json['username'] as String,
      password: json['password'] as String,
      firstName: json['firstName'] as String?,
      lastName: json['lastName'] as String?,
      publicKey: json['publicKey'] as String?,
      privateKeyEncrypted: json['privateKeyEncrypted'] as String?,
      salt: json['salt'] as String?,
      email: json['email'] as String?,
    );
  }

  Map<String, dynamic> toJson() => {
    'username': username,
    'password': password,
    if (firstName != null) 'firstName': firstName,
    if (lastName != null) 'lastName': lastName,
    if (publicKey != null) 'publicKey': publicKey,
    if (privateKeyEncrypted != null) 'privateKeyEncrypted': privateKeyEncrypted,
    if (salt != null) 'salt': salt,
    if (email != null) 'email': email,
  };
}
